src/Kiota.Builder/PathSegmenters/GoPathSegmenter.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections.Generic;
+using System.Linq;
+using Kiota.Builder.Extensions;
+
+namespace Kiota.Builder {
+    public class GoPathSegmenter : CommonPathSegmenter
+    {
+        public GoPathSegmenter(string rootPath, string clientNamespaceName) : base(rootPath, clientNamespaceName) {}
+        public override string GetFileSuffix(CodeElement currentElement)
+        {
+            switch (currentElement) {
+                case CodeNamespace n:
```

Comment:
It should be okay to remove the variable for now as it is unused. 
```suggestion
                case CodeNamespace:
```

Reply:
fixed, I think I just when with Github copilot suggestion for this one :)